solr-ingest/src/main/java/edu/unc/lib/dl/data/ingest/solr/filter/SetContentTypeFilter.java

```diff
@@ -0,0 +1,147 @@
+                    "mimetypeToExtension.txt")));
+            contentTypeProperties = new Properties();
+            contentTypeProperties.load(new InputStreamReader(this.getClass().getResourceAsStream(
+                    "toContentType.properties")));
+        } catch (IOException e) {
+            log.error("Failed to load mimetype mappings", e);
+        }
+    }
+
+    @Override
+    public void filter(DocumentIndexingPackage dip) throws IndexingException {
+        BinaryObject binObj = getFileObject(dip).getOriginalFile();
+        String filepath = binObj.getFilename();
+        String mimetype = binObj.getMimetype();
+        log.debug("The binary has filepath " + filepath + " and mimetype " + mimetype);
+        List<String> contentTypes = new ArrayList<String>();
+        extractContentType(filepath, mimetype, contentTypes);
+        dip.getDocument().setContentType(contentTypes);
+    }
+
+    private FileObject getFileObject(DocumentIndexingPackage dip) throws IndexingException {
+        ContentObject obj = dip.getContentObject();
+        FileObject fileObj;
+        if (obj instanceof WorkObject) {
+            fileObj = ((WorkObject) obj).getPrimaryObject();
+        } else {
+            fileObj = (FileObject) obj;
+        }
+        return fileObj;
+    }
+
+    private String getExtension(String filepath, String mimetype) {
+        if (filepath != null) {
+            Matcher matcher = EXTENSION_REGEX.matcher(filepath);
+            if (matcher.matches()) {
+                String extension = matcher.group(1);
+                if (extension.length() <= EXTENSION_LIMIT) {
+                    return extension.toLowerCase();
+                }
+            }
+        }
+        if (mimetype != null) {
+            String extension = mimetypeToExtensionMap.getProperty(mimetype);
+            return extension;
+        }
+        return null;
+    }
+
+    private void extractContentType(String filepath, String mimetype, List<String> contentTypes) {
+        ContentCategory contentCategory = getContentCategory(mimetype, getExtension(filepath, mimetype));
+        // add string with name + display-name to list of content types
+        contentTypes.add('^' + contentCategory.getJoined());
+        StringBuilder contentType = new StringBuilder();
+        contentType.append('/').append(contentCategory.name()).append('^');
+        String extension = getExtension(filepath, mimetype);
+        if (extension == null) {
+            contentType.append("unknown,unknown");
+        } else {
+            contentType.append(extension).append(',').append(extension);
+        }
+        // add string with content category name + extension to list of content types
+        contentTypes.add(contentType.toString());
+    }
+
+    private ContentCategory getContentCategory(String mimetype, String extension) {
+        if (mimetype == null) {
+            return ContentCategory.unknown;
+        }
+        int index = mimetype.indexOf('/');
```

Comment:
@ben I'm wondering whether we can discard lines 126-139 altogether, and also get rid of the null check on 141, and just handle all cases the same way. Is there any reason to treat the image/audio/video cases as special ones?